pgmpy/inference/ExactInference.py

```diff
@@ -228,6 +230,47 @@ def induced_graph(self, elimination_order):
         >>> inference.induced_graph(['C', 'D', 'A', 'B', 'E'])
         <networkx.classes.graph.Graph at 0x7f34ac8c5160>
         """
+        # If the elimination_order list is incomplete, raise an error
+        if len(elimination_order) < len(self.variables):
+            raise ValueError("Elimination order incomplete")
+        # If the elimination_order list is incomplete, raise an error
+        if set(elimination_order) != set(self.variables):
+            raise ValueError("Set of variables in elimination order"
+                             " different from variables in model")
+
+        eliminated_variables = set()
+        working_factors = {node: [factor.scope() for factor in self.factors[node]]
+                           for node in self.factors}
+
+        # The set of cliques should be in the induced graph
+        cliques = set()
+        for factors in working_factors.values():
+            for factor in factors:
+                cliques.add(tuple(factor))
+
+        # Removing all the factors containing the variables which are
+        # eliminated (as all the factors should be considered only once)
+        for var in elimination_order:
+            factors = [factor for factor in working_factors[var]
+                       if not set(factor).intersection(eliminated_variables)]
+            phi = set(itertools.chain(*factors)).difference({var})
+            cliques.add(tuple(phi))
+            del working_factors[var]
+            for variable in phi:
+                working_factors[variable].append(list(phi))
+            eliminated_variables.add(var)
+
+        edges = []
+        # add edges corresponding to each clique
+        for clique in filter(lambda x: len(x) > 1, cliques):
+            for i, j in itertools.combinations(clique, 2):
+                edges.append((i, j))
+
+        # Final induced graph
+        graph = nx.Graph()
+        graph.add_edges_from(edges)
```

Comment:
@pratyakshs No need to first create a graph object and then adding edges. You could simply do `return nx.Graph(edges)`